docs/modules/ROOT/pages/references/style-guide.adoc

```diff
@@ -0,0 +1,416 @@
+----
+// CORRECT
+local foo = [ "a", "b" ] ;
+local bar = [
+  "a",
+  "b",
+];
+
+// INCORRECT
+local baz = ["a", "b"];
+----
+
+* Start objects on the same line as the variables they're assigned to.
++
+[source,jsonnet]
+----
+// PREFERRED
+local foo = { hello: "world" };
+local bar = {
+  hello: "world",
+};
+
+// ACCEPTABLE
+local baz =
+  {
+    hello: "world",
+  };
+----
+
+* Objects within a conditional start on the same line as the condition.
++
+[source,jsonnet]
+----
+// PREFERRED
+local foo(x) =
+  if x == 42 then {
+    result: "The Answer",
+  }
+  else {
+    result: "Don't know",
+  };
+
+
+// ACCEPTABLE
+local bar(x) =
+  if x == 42 then
+    {
+      result: "The Answer",
+    }
+  else
+    {
+      result: "Don't know",
+    };
+----
+
+* Start `if` and `else` on new lines and prefer to keep `else if` together.
++
+[sourece,jsonnet]
+----
+// PREFERRED
+local foo(x) =
+  if x < 42 then {
+    result: "No enought",
+  }
+  else if > 42 then {
+    result: "Too much",
+  }
+  else {
+    result: "The Answer",
+  }
+
+// ACCEPTABLE
+local bar(x) =
+  if x < 42 then {
+    result: "No enought",
+  }
+  else
+    if > 42 then {
+      result: "Too much",
+    }
+    else {
+      result: "The Answer",
+    }
+----
+
+* Omit tailing `,` on single line arrays and objects.
+  Keep them when splitting over multiple lines.
++
+[source,jsonnet]
+----
+// CORRECT
+local a = [ "a", "b" ] ;
+local b = { hello: "world" };
+local c =
+  [
+    "a",
+    "b",
+  ];
+local d =
+  {
+    hello: "world",
+  };
+
+// INCORRECT
+local e = [ "a", "b", ];
+local f = { hello: "world", };
+----
+
+* Use 2-space indentation in general.
+
+* Only function parameter declarations use 4-space indentation, to visually differentiate parameters from function body.
++
+[source,jsonnet]
+----
+// CORRECT
+local multiply(
+    number1,
+    number2) =
+  {
+    result: number1 * number 2
+  }
+----
+
+* Omit vertical alignment.
+  They draw attention to the wrong parts of the code and result in hard to review pull requests due to the white space changes.
++
+[source,jsonnet]
+----
+// Don't align vertically
+local plus     = "+";
+local minus    = "-";
+local multiply = "*";
+
+// Do the following
+local plus = "+";
+local minus = "-";
+local multiply = "*";
+----
+
+=== Blank lines (vertical whitespace)
+
+* A single blank line appears:
+** Within functions bodies, as needed to create logical groupings of statements.
+** Optionally before the first member or after the last member of a template or function.
+* Use one or two blank line(s) to separate logical blocks in files.
+  Those blocks can be single function definitions or groups of local variables that semantically belong together.
+* Excessive number of blank lines is discouraged.
+
+=== Defining and using abstractions
+
+==== Defining templates
+
+* Rather than defining a concrete JSON object, it's often useful to define a template which takes some set of parameters before being materialized into JSON.
```

Comment:
```suggestion
* Rather than defining a concrete JSON object, it's often useful to define a template which takes a set of parameters.
  Such templates can be used to parametrize JSON objects that need to be materialized multiple times with only small changes.
```